internal/aks/create.go

```diff
@@ -0,0 +1,217 @@
+		})
+
+	return err
+}
+
+// createOrUpdateCluster creates a new managed Kubernetes cluster
+func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient *containerservice.ManagedClustersClient, spec *aksv1.AKSClusterConfigSpec) (c containerservice.ManagedCluster, err error) {
+
+	dnsPrefix := spec.DNSPrefix
+	if dnsPrefix == nil {
+		dnsPrefix = to.StringPtr(spec.ClusterName)
+	}
+
+	tags := make(map[string]*string)
+	for key, val := range spec.Tags {
+		if val != "" {
+			tags[key] = to.StringPtr(val)
+		}
+	}
+	clusterName := spec.ClusterName
+	tags["displayName"] = to.StringPtr(clusterName)
+
+	kubernetesVersion := spec.KubernetesVersion
+
+	var vmNetSubnetID *string
+	networkProfile := &containerservice.NetworkProfile{}
+	if hasCustomVirtualNetwork(spec) {
+		virtualNetworkResourceGroup := spec.ResourceGroup
+
+		//if virtual network resource group is set, use it, otherwise assume it is the same as the cluster
+		if spec.VirtualNetworkResourceGroup != nil {
+			virtualNetworkResourceGroup = *spec.VirtualNetworkResourceGroup
+		}
+
+		vmNetSubnetID = to.StringPtr(fmt.Sprintf(
+			"/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Network/virtualNetworks/%v/subnets/%v",
+			spec.SubscriptionID,
+			virtualNetworkResourceGroup,
+			spec.VirtualNetwork,
+			spec.Subnet,
+		))
+
+		networkProfile.DNSServiceIP = spec.NetworkDNSServiceIP
+		networkProfile.DockerBridgeCidr = spec.NetworkDockerBridgeCIDR
+		networkProfile.ServiceCidr = spec.NetworkServiceCIDR
+
+		if spec.NetworkPlugin != nil {
+			networkProfile.NetworkPlugin = containerservice.NetworkPlugin(*spec.NetworkPlugin)
+		} else {
+			networkProfile.NetworkPlugin = containerservice.Kubenet
+		}
+
+		// if network plugin is 'Azure', set PodCIDR
+		if networkProfile.NetworkPlugin == containerservice.Azure {
+			networkProfile.PodCidr = spec.NetworkPodCIDR
+		}
+
+		if spec.LoadBalancerSKU != nil {
+			loadBalancerSku := containerservice.LoadBalancerSku(*spec.LoadBalancerSKU)
+			networkProfile.LoadBalancerSku = loadBalancerSku
+		}
+
+		if spec.NetworkPolicy != nil {
+			networkProfile.NetworkPolicy = containerservice.NetworkPolicy(*spec.NetworkPolicy)
+		}
+	}
+
+	var agentPoolProfiles []containerservice.ManagedClusterAgentPoolProfile
+	if hasAgentPoolProfile(spec) {
+		for _, np := range spec.NodePools {
+			if np.OrchestratorVersion == nil {
+				np.OrchestratorVersion = spec.KubernetesVersion
+			}
+			agentProfile := containerservice.ManagedClusterAgentPoolProfile{
+				Name:                np.Name,
+				Count:               np.Count,
+				MaxPods:             np.MaxPods,
+				OsDiskSizeGB:        np.OsDiskSizeGB,
+				OsType:              containerservice.OSType(np.OsType),
+				VMSize:              containerservice.VMSizeTypes(np.VMSize),
+				Mode:                containerservice.AgentPoolMode(np.Mode),
+				OrchestratorVersion: np.OrchestratorVersion,
+			}
+			if np.AvailabilityZones != nil {
+				agentProfile.AvailabilityZones = np.AvailabilityZones
```

Comment:
Could be set in the instantiation on the lines above instead of checking nil and setting it here

Reply:
ok, done